discovery-provider/src/tasks/users.py

```diff
@@ -415,6 +424,47 @@ def update_user_associated_wallets(
         )
 
 
+class UserEventsMetadata(TypedDict):
+    referrer: int
+    is_mobile_user: bool
+
+
+def update_user_events(
+    session: Session,
+    user_record: User,
+    events: UserEventsMetadata
+) -> None:
+    """Updates the user events table"""
+    try:
+        if not isinstance(events, dict):
+            # There is something wrong with events, don't process it
+            return
+
+        # Mark existing UserEvents entries as not current
+        session.query(UserEvents).filter_by(user_id=user_record.user_id).update(
+            {"is_current": False}
+        )
+
+        user_events = UserEvents(
+            user_id=user_record.user_id,
+            is_current=True,
+            blocknumber=user_record.blocknumber,
+            blockhash=user_record.blockhash,
+        )
+        for event, value in events.items():
+            if event == 'referrer':
+                user_events.referrer = value
+            elif event == 'is_mobile_user':
+                user_events.is_mobile_user = value
```

Comment:
Could also check type on `value` - that it's bool for is_mobile_user and int for referrer - but also should be checked in parsing JSON - so not necessary

Reply:
typecasts are event better :)